test/e2e/failover_test.go

```diff
@@ -141,6 +141,117 @@ var _ = framework.SerialDescribe("failover testing", func() {
 			})
 		})
 	})
+
+	ginkgo.Context("Deployment status synchronization when cluster failed and recovered soon", func() {
```

Comment:
This e2e test doesn't involve failover. It's more like a test case for resource state collection. How about moving it to `resource_test.go`?